Docs: quick start for synapse SDK

longfeiWan9 opened the pull request:
Closes #323 

This PR improves the Quick Start guide to provide a better developer onboarding experience.

**🚀 Enhanced Quick Start Guide**
- **Restructured into 5 clear steps**: Initialize SDK → Get Test Tokens → USDFC Deposit → Payment Setup → Store & Download
- **Added test token acquisition**: Instructions for getting tFIL and USDFC from faucets
- **Improved code examples**: Complete working examples with proper error handling and callbacks
- **Added "What's Next?" section**: Showcases advanced features and provides clear paths for deeper learning
- **Removed the usage, API** and rest of section out. To keep quick start simple, those content will go into developer guide section.

timfong888 reviewed (changes requested):
small changes....

hugomrdias commented on docs/src/content/docs/getting-started/quick-start.mdx:
```diff
@@ -0,0 +1,229 @@
+**Get test USDFC tokens:**
+- Visit the [Filecoin Calibration USDFC Faucet](https://forest-explorer.chainsafe.dev/faucet/calibnet_usdfc)
+- Enter your wallet address to receive free test USDFC tokens
+- You need USDFC for storage payments
```
you can just do https://synapse.filecoin.services/api/filoz/synapse-core/utils/functions/claimtokens/

timfong888 replied:
Does the quick start *only* use calibration net?  Could we:

a) Include one for mainnet, that's the primary path we want and have as an option a way to go on testnet (I know that testnet sounds like the place to do development, but I think getting started should be on mainnet, wdyt?  The cost is relatively low and I'd like to get an active dapp on mainnet.  

b) can we incorporate this end point from @hugomrdias ?

hugomrdias commented on docs/src/content/docs/getting-started/quick-start.mdx:
```diff
@@ -0,0 +1,229 @@
+  ethers.parseUnits('10', 18),   // Rate allowance: 10 USDFC per epoch
+  ethers.parseUnits('1000', 18), // Lockup allowance: 1000 USDFC total
+  86400n                         // Max lockup period: 30 days (in epochs)
+)
```
we are defaulting allowances to maxUint256 for some time now

longfeiWan9 commented:
Thank you for reviewing and providing comments. @timfong888 @Kubuxu @hugomrdias 

Since Nick created a new PR for all docs, I have addressed all the comments for `quick start` over there. So I will close this PR and focus on #404
